app/uk/gov/hmrc/slacknotifications/connectors/UserManagementConnector.scala

```diff
@@ -16,157 +16,78 @@
 
 package uk.gov.hmrc.slacknotifications.connectors
 
-import play.api.cache.AsyncCacheApi
-import play.api.libs.functional.syntax.unlift
-
-import javax.inject.{Inject, Singleton}
 import play.api.Logging
-import play.api.libs.json.{Format, Json}
-import uk.gov.hmrc.http.{HeaderCarrier, HttpReads, StringContextOps}
-import uk.gov.hmrc.http.client.HttpClientV2
-import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
-import uk.gov.hmrc.slacknotifications.config.UserManagementAuthConfig
-
 import play.api.libs.functional.syntax._
 import play.api.libs.json._
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, HttpReads, StringContextOps}
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import uk.gov.hmrc.slacknotifications.connectors.UserManagementConnector.TeamName.teamNameReads
 
+import java.net.URL
+import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class UserManagementConnector @Inject()(
   httpClientV2  : HttpClientV2,
-  servicesConfig: ServicesConfig,
-  authConfig    : UserManagementAuthConfig,
-  tokenCache    : AsyncCacheApi
+  servicesConfig: ServicesConfig
 )(implicit ec: ExecutionContext) extends Logging {
 
-  import UserManagementConnector._
-  import authConfig._
   import HttpReads.Implicits._
+  import UserManagementConnector._
 
-  private val url: String = {
-    val keyInServices = "user-management.url"
-    servicesConfig.getConfString(keyInServices, throw new RuntimeException(s"Could not find config $keyInServices"))
-  }
-
-  private lazy val loginUrl = {
-    val keyInServices = "user-management.loginUrl"
-    servicesConfig.getConfString(keyInServices, throw new RuntimeException(s"Could not find config $keyInServices"))
-  }
+  private val baseUrl = servicesConfig.baseUrl("user-management")
 
-  def login(): Future[UmpToken] = {
-    implicit val lrf: OFormat[UmpLoginRequest] = UmpLoginRequest.format
-    implicit val atf: OFormat[UmpAuthToken]    = UmpAuthToken.format
-    implicit val hc: HeaderCarrier             = HeaderCarrier()
-    for {
-      token <- httpClientV2.post(url"$loginUrl")
-                 .withBody(Json.toJson(UmpLoginRequest(username, password)))
-                 .execute[UmpAuthToken]
-      _      = logger.info("logged into UMP")
-    } yield token
+  private def getUser(endpoint: String, userType: String, username: String)(implicit hc: HeaderCarrier): Future[List[TeamName]] = {
+    implicit val teamsAndRolesReads = {
+      implicit val tnr = teamNameReads
+      Reads.at[List[TeamName]](__ \ "teamsAndRoles")
+    }
+    httpClientV2
+      .get(new URL(s"$baseUrl/$endpoint"))
```

Comment:
Could you not use .get(url"$baseUrl/$endpoint")  without the need for new URL ... perhaps more consistent with HttpClientV2